src/JustSaying.Models/Message.cs

```diff
@@ -6,12 +6,14 @@ public abstract class Message
     {
         protected Message()
         {
-            TimeStamp = DateTime.UtcNow;
-            Id = Guid.NewGuid();
+            TimeStampWithOffset = DateTimeOffset.Now;
```

Comment:
There's a problem here.  If this is used as a base class on the receiving end and the publisher was using a version that didn't have the property, the ctor is going to set the value to Now and leave it like that.

I guess the publisher should set it if its null.  And for those features that don't use the JS publisher, their local publishers will just need to do likewise.

Reply:
Perhaps we could leave it null in the constructor, and populate it in the library before we send it? That way we keep the init out of the Models lib, and provides a possible extension point for what you were thinking of with other users possibly using a different 'base' TZ.